rpmlint/checks/BuildDateCheck.py

```diff
@@ -0,0 +1,29 @@
+import re
+import stat
+import time
+
+from rpmlint.checks.AbstractCheck import AbstractFilesCheck
+
+
+class BuildDateCheck(AbstractFilesCheck):
+    def __init__(self, config, output):
+        super().__init__(config, output, r'.*')
+        self.looksliketime = re.compile('(2[0-3]|[01]?[0-9]):([0-5]?[0-9]):([0-5]?[0-9])')
+        self.istoday = re.compile(time.strftime('%b %e %Y'))
+
+    def check_file(self, pkg, filename):
+        if filename.startswith('/usr/lib/debug') or pkg.isSource():
+            return
+
+        if not stat.S_ISREG(pkg.files()[filename].mode):
+            return
+
+        grep_date = pkg.grep(self.istoday, filename)
+
+        if len(grep_date):
+            grep_time = pkg.grep(self.looksliketime, filename)
+
+            if len(grep_time):
+                self.output.add_info('W', pkg, 'file-contains-date-and-time', filename)
+            else:
+                self.output.add_info('W', pkg, 'file-contains-current-date', filename)
```

Comment:
I would even error out, these when happen are so much pain in the arse for rebuilds...